tests/test_sanity_sample.py

```diff
@@ -291,20 +295,34 @@ def test_pretrained_model_train(config, tmp_path, multiprocessing_distributed, c
         args["--cpu-only"] = True
     elif multiprocessing_distributed:
         args["--multiprocessing-distributed"] = True
+    elif config['nncf_config']["model"] == "inception_v3":
+        pytest.skip("InceptionV3 may not be trained in DataParallel "
+                    "because it outputs namedtuple, which DP seems to be unable "
+                    "to support even still.")
 
     runner = Command(create_command_line(args, config["sample_type"]))
     runner.run()
     last_checkpoint_path = os.path.join(checkpoint_save_dir, get_name(config_factory.config) + "_last.pth")
     assert os.path.exists(last_checkpoint_path)
-    assert torch.load(last_checkpoint_path)['compression_level'] in (CompressionLevel.FULL, CompressionLevel.PARTIAL)
+    if 'compression' in config['nncf_config']:
+        allowed_compression_levels = (CompressionLevel.FULL, CompressionLevel.PARTIAL)
+    else:
+        allowed_compression_levels = (CompressionLevel.NONE,)
+    assert torch.load(last_checkpoint_path)['compression_level'] in allowed_compression_levels
+
 
+def depends_on_pretrained_train(request, test_case_id: str, current_multiprocessing_distributed: bool):
+    full_test_case_id = test_case_id + ('-distributed' if current_multiprocessing_distributed else '-dataparallel')
+    primary_test_case_name = f'test_pretrained_model_train[{full_test_case_id}]'
+    depends(request, [primary_test_case_name])
 
+
+@pytest.mark.dependency()
 @pytest.mark.parametrize(
-    "multiprocessing_distributed", [
-        pytest.param(True, marks=pytest.mark.dependency(depends=["train_distributed"])),
-        pytest.param(False, marks=pytest.mark.dependency(depends=["train_dataparallel"]))],
```

Comment:
Note that this did not actually set up dependencies between the test cases corresponding to one and the same model-dataset-execution type configuration.